### PR TITLE
Add alias annotations before the stereotypes

### PR DESCRIPTION
**Pull request by dstepanov**
Fixes https://github.com/micronaut-projects/micronaut-core/issues/9919

### CODE REVIEW

File: inject-java/src/test/groovy/io/micronaut/visitors/PropertyElementSpec.groovy [hunk 1]
```diff
@@ -278,19 +278,15 @@ class Parameters {
 
         def beanProperty = parameter.type.beanProperties.get(0)
         beanProperty.type.annotationNames.sort() == [
```
Comment (dstepanov) [inline inject-java/src/test/groovy/io/micronaut/visitors/PropertyElementSpec.groovy:280]: This test had to be change becase `BeanProperties` is now added as a stereotype.

File: inject-java/src/test/groovy/io/micronaut/inject/configproperties/InheritedConfigurationReaderPrefixSpec.groovy [hunk 2]
```diff
@@ -72,7 +73,7 @@ class MyBean  {
             beanDefinition.getInjectedMethods()[0].name == 'setMyValue'
             def metadata = beanDefinition.getInjectedMethods()[0].getAnnotationMetadata()
             metadata.hasAnnotation(Property)
-            metadata.getValue(Property, "name", String).get() == 'endpoints.my-value'
+            metadata.getValue(Property, "name", String).get() == 'simple.my-value'
```
Comment (dstepanov) [inline inject-java/src/test/groovy/io/micronaut/inject/configproperties/InheritedConfigurationReaderPrefixSpec.groovy:76]: This scenario was tested incorrectly and this should be the correct output

File: inject-groovy/src/test/groovy/io/micronaut/inject/configproperties/InheritedConfigurationReaderPrefixSpec.groovy [hunk 3]
```diff
@@ -56,7 +57,7 @@ class MyBean  {
         beanDefinition.getInjectedMethods()[0].name == 'setMyValue'
         def metadata = beanDefinition.getInjectedMethods()[0].getAnnotationMetadata()
         metadata.hasAnnotation(Property)
-        metadata.getValue(Property, "name", String).get() == 'endpoints.my-value'
+        metadata.getValue(Property, "name", String).get() == 'simple.my-value'
```
Comment (dstepanov) [inline inject-groovy/src/test/groovy/io/micronaut/inject/configproperties/InheritedConfigurationReaderPrefixSpec.groovy:60]: This scenario was tested incorrectly and this should be the correct output

**Comment by graemerocher** [comment]
test failures?

**Comment by dstepanov** [comment]
Had to fix another KSP problem

**Review by graemerocher — APPROVED** [review]

**Pull request merged**